src/entity/EntityFactory.php

```diff
@@ -171,6 +173,19 @@ public function __construct(){
 			return new SplashPotion(Helper::parseLocation($nbt, $world), null, $potionType, $nbt);
 		}, ['ThrownPotion', 'minecraft:potion', 'thrownpotion']);
 
+		$this->register(Trident::class, function(World $world, CompoundTag $nbt) : Trident{
+			$itemTag = $nbt->getCompoundTag(Trident::TAG_ITEM);
+			if($itemTag === null){
+				throw new SavedDataLoadingException("Expected \"" . Trident::TAG_ITEM . "\" NBT tag not found");
+			}
+
+			$item = Item::nbtDeserialize($itemTag);
+			if($item->isNull() || !$item instanceof TridentItem){
```

Comment:
Is there a reason to require this to be a TridentItem? It doesn't look like it needs any specific methods.

Reply:
Consistency, if I do `TridentEntity->getItem()` I would expect a `TridentItem`.